### ISSUE TITLE
configure errors on solaris 11.2 /bin/sh

### ISSUE THREAD

**Issue by pavanbalaji** [pmodels/mpich#3911]
Error reported by Paul Hargrove:

below are at least 2 distinct errors from configure with the Solaris 11.2 /bin/sh

The -c option to head is not in POSIX: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/head.html.

Not sure what the other one is about.

-Paul

```
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
head: illegal option -- c
./maint/gentests_dtp.sh[65]: [: head: unknown operator
config.status: executing collcvartests commands
./maint/test_coll_algos.sh: line 642: syntax error at line 649: ``' unmatched
config.status: executing depfiles commands
```

**Comment by pavanbalaji** [comment]
Cc'ing @PHHargrove.

**Comment by gcongiu** [comment]
`head -c 1` matches the first character in the line to see if that is a `#` (comment line).

**Comment by PHHargrove** [comment]
The second error regarding an unmatched backtick in maint/test_coll_algos.sh also seen with the /bin/sh in AIX 7.1.  The man page confirms AIX is OK with `head -c 1`

**Comment by hzhou** [comment]
Responding to the unmatched backtick -- we just leaked this typo through. It is strange that `bash` and `sh` on FreeBSD don't mind these syntax errors.

**Comment by hzhou** [comment]
@gcongiu Maybe you should try `cut -c1`?

**Comment by gcongiu** [comment]
@hzhou I know, I just didn't realize this would cause issues on Solaris. Shell is convenient for simple tasks, if python was an option I would use that. In any case perl will never have me :p

**Comment by raffenet** [comment]
@gcongiu @hzhou this can be closed, right?

**Comment by hzhou** [comment]
`head -c 1` is still in `test/mpi/maint/gentests_dtp.sh`. @gcongiu would you like to fix it?

**Comment by gcongiu** [comment]
@hzhou sure I will fix it by the end of this week

**Comment by raffenet** [comment]
Fixed in #4064
